code/tika/TikaRestClient.php

```diff
@@ -5,6 +5,29 @@
 
 class TikaRestClient extends Client
 {
+    /**
+     * Authentication options to be sent to the Tika server
+     *
+     * @var array
+     */
+    protected $options = array('username' => null, 'password' => null);
+
+    /**
+     * @var array
+     */
+    protected $mimes = array();
+
+    public function __construct($baseUrl = '', $config = null)
+    {
+        if (defined('SS_TIKA_USERNAME') && defined('SS_TIKA_PASSWORD')) {
+            $this->options = array(
+                'username' => SS_TIKA_USERNAME,
```

Comment:
As an update to this, we should probably support `Environment::getEnv()` equivalent as well, since this is the preferred configuration mechanism.

No reason we can't support constants either though, it's just less preferred.

Reply:
For the 4.0 compat version, rather than this, I should say. :)